WebDriverAgentLib/Utilities/FBPredicate.m

```diff
@@ -9,13 +9,20 @@
 
 #import "FBPredicate.h"
 
+#import "FBXCodeCompatibility.h"
+
 @implementation FBPredicate
 
 + (NSPredicate *)predicateWithFormat:(NSString *)predicateFormat,  ...
 {
   va_list args;
   va_start(args, predicateFormat);
   NSPredicate *predicate = [NSPredicate predicateWithFormat:predicateFormat arguments:args];
+  va_end(args);
+  // SDK 11+ does not require the hack to be applied
+  if (XCUIElement.fb_isSdk11SnapshotApiSupported) {
```

Comment:
After opening 'com.apple.Maps', I sent below query. Then, I was not able to find an element.

- before
    - > `@driver.find_element :name, 'Tracking'`
        - `#<Selenium::WebDriver::Element:0x..fc7f7c13d2748cb5c id="6D000000-0000-0000-C403-000000000000">`
    - > `@driver.find_element :predicate, 'name == "Tracking"'`
        - `#<Selenium::WebDriver::Element:0x6b846bfb7e19e3f0 id="8C000000-0000-0000-D203-000000000000">`
- after
    - > `@driver.find_element :predicate, 'name == "Tracking"'`
        - `Selenium::WebDriver::Error::NoSuchElementError: An element could not be located on the page using the given search parameters.
from NoSuchElementError: An element could not be located on the page using the given search parameters.`

Xcode 11.2.1, iOS 13.2 real device

---

When I removed this condition, the `tracking` was returned without such regression

Reply:
strange. I'll put it back for now. Perhaps, we can still implement it more precisely.